src/core/http-response/http-response-impl.ts

```diff
@@ -0,0 +1,59 @@
+import { HttpResponse, HttpBody, HttpHeaders, HttpStatus } from "../http";
+
+export class HttpResponseImpl implements HttpResponse {
+  constructor(
+    public readonly status: HttpStatus,
+    public readonly body: HttpBody,
+    public readonly headers: HttpHeaders
+  ) {}
+
+  setHeaders(headers: HttpHeaders) {
```

Comment:
Is there any way that we can create an abstract class of HttpMessage so that we don't duplicate some of the accessors(addHeader, removeHeader)?

Reply:
Maybe not a super class or abstract class but a set of functions that return `headers`. I will create an issue for this once we merged this PR (because I want to point to the code in the issue).